src/eventbridge/eventpattern/index.ts

```diff
@@ -0,0 +1,798 @@
+import { FunctionDecl } from "../../declaration";
+import {
+  BinaryExpr,
+  CallExpr,
+  ElementAccessExpr,
+  Expr,
+  PropAccessExpr,
+  UnaryExpr,
+} from "../../expression";
+import {
+  assertDefined,
+  assertNever,
+  assertNumber,
+  assertString,
+} from "../../assert";
+import { isReturn, Stmt } from "../../statement";
+import {
+  PatternDocument,
+  Pattern,
+  isPatternDocument,
+  isNumericAggregationPattern,
+  isNumericRangePattern,
+  isPresentPattern,
+  isAggregatePattern,
+  isAnythingButPattern,
+  isExactMatchPattern,
+  isPrefixMatchPattern,
+  isEmptyPattern,
+  patternDocumentToEventPattern,
+} from "./pattern";
+import * as fnls_event_bridge from "./types";
+import {
+  joinNumericRange,
+  reduceNumericAggregate,
+  mergeNumericOr,
+  negateNumericRange,
+  andNumericAggregation,
+  andNumericAggregationWithRange,
+  createSingleNumericRange,
+} from "./numeric";
+import {
+  BinaryOp,
+  isBinaryExpr,
+  isBooleanLiteral,
+  isCallExpr,
+  isElementAccessExpr,
+  isIdentifier,
+  isNullLiteralExpr,
+  isNumberLiteralExpr,
+  isPropAccessExpr,
+  isStringLiteralExpr,
+  isUnaryExpr,
+} from "../..";
+import {
+  INCLUDES_SEARCH_ELEMENT,
+  OPERATIONS,
+  STARTS_WITH_SEARCH_STRING,
+} from "./constants";
+import { invertBinaryOperator } from "./utils";
+
+/**
+ * https://github.com/sam-goodwin/functionless/issues/37#issuecomment-1066313146
+ */
+export const synthesizeEventPattern = (
```

Comment:
Is it good practice to have logic in `index.ts`? Took me a while to find this funciton. Would prefer a file, `synth.ts`

Reply:
I've always used the `index.ts` to define the top level module. The same as if it as `eventpattern.ts`. I am fine with extracting it out into a new fine with explicit `index.ts` exports.